Sources/ArgumentParser/Usage/UsageGenerator.swift

```diff
@@ -114,29 +114,11 @@ extension ArgumentDefinition {
   }
   
   var sortedNames: [Name] {
-    return names
-      .sorted { (lhs, rhs) -> Bool in
-        switch (lhs, rhs) {
-        case let (.long(l), .long(r)):
-          return l < r
-        case (_, .long):
-          return true
-        case (.long, _):
-          return false
-        case let (.short(l), .short(r)):
-          return l < r
-        case (_, .short):
-          return true
-        case (.short, _):
-          return false
-        case let (.longWithSingleDash(l), .longWithSingleDash(r)):
-          return l < r
-        }
-    }
+    return names.filter{ $0 == .short($0.valueString.first!) } + names.filter{ $0 != .short($0.valueString.first!) }
```

Comment:
Two notes here:

- Since we're breaking this array of names into two groups, first all the short names and then all the non-short names, let's pull the test for "this name is short" out into a property on `Name`.
- The returned array is no longer sorted, so we should change the name of the property as well. Alternatively, we could just put them in this order at initialization and get rid of the property altogether.

Reply:
* I am not sure about what you mean by this "let's pull the test for "this name is short" out into a property on Name.". Sorry about my English. Do you mean something like  "Name.isShort" ? (If yes, Open for naming suggestions)

* I can rename it. "partitionedNames" maybe?. Alternative sounds good, too. But I am not sure about changing the original order.

Reply:
`Name.isShort` would be great! And I think `partitionedNames` is better than `sortedNames` — hard to find a good name for this concept, exactly.